web/src/narrow_banner.ts

```diff
@@ -124,12 +124,28 @@ export function pick_empty_narrow_banner(): NarrowBannerData {
                   },
               ),
     };
+    const combined_feed_banner = {
+        title: $t({defaultMessage: "There are no messages in your combined feed."}),
+        // Spectators cannot start a conversation.
+        html: page_params.is_spectator
+            ? ""
+            : $t_html(
+                  {
+                      defaultMessage:
+                          "Would you like to <z-link>view messages in all public channels</z-link>?",
+                  },
+                  {
+                      "z-link": (content_html) =>
+                          `<a href="#narrow/streams/public">${content_html.join("")}</a>`,
+                  },
+              ),
+    };
     const default_banner_for_multiple_filters = $t({defaultMessage: "No search results."});
 
     const current_filter = narrow_state.filter();
 
     if (current_filter === undefined || current_filter.is_in_home()) {
```

Comment:
An `undefined` filter means that we'd either be in the inbox view or recent conversations view, but those view's don't use this function/module.

I think these should probably be split out so that the undefined case still returns our general default (even though that will never happen), maybe with a comment. And then the `is_in_home` case would return the special case of the empty combined feed banner. 

That way you don't need to make it a constant either since it's just used in that one case.